runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WindowingStrategyTranslation.java

```diff
@@ -406,4 +416,64 @@ public static SdkFunctionSpec toProto(WindowFn<?, ?> windowFn, SdkComponents com
           e);
     }
   }
+
+  @AutoValue
+  abstract static class RawWindowFn<T, W extends BoundedWindow> extends WindowFn<T, W> {
+    static <T, W extends BoundedWindow> WindowFn<T, W> forFunctionSpec(
+        SdkFunctionSpec functionSpec) {
+      return new AutoValue_WindowingStrategyTranslation_RawWindowFn(functionSpec);
+    }
+
+    abstract SdkFunctionSpec getFunctionSpec();
+
+    @Override
+    public Collection<W> assignWindows(AssignContext c) {
+      throw new UnsupportedOperationException();
```

Comment:
Error messages

Reply:
Done, although I don't find them particularly useful for this type of exception.